include/slam_toolbox/map_saver.hpp

```diff
@@ -32,7 +32,19 @@ namespace map_saver
 class MapSaver
 {
 public:
-  MapSaver(rclcpp::Node::SharedPtr node, const std::string & service_name);
+  MapSaver(
+    rclcpp::node_interfaces::NodeBaseInterface::SharedPtr base_interface,
```

Comment:
Ditto, I'll stop remarking on this now. Just template by the NodeT and call it a day